### PR TITLE
make .schedule and .config properly support and document ScheduledBoredCommandsJobs

### PR DESCRIPTION
**Pull request by Ixrec**
oversights in #390

### CODE REVIEW

File: Izzy-Moonbot/Modules/ModMiscModule.cs [hunk 1]
```diff
@@ -574,6 +575,16 @@ public async Task ScheduleCommandAsync([Remainder]string argsString = "")
...
                         break;
+                    case "ScheduledBoredCommandsJob":
+                        content = "**Bored Commands**\n" +
+                                  "*If no one has posted in `BoredChannel` within the last `BoredCooldown` seconds, this job posts one of the strings in `BoredCommands`, " +
+                                      "typically a command meant to be executed by Izzy herself or another bot. Then reschedules itself for `BoredCooldown` seconds in the future.*\n" +
+                                  ":warning: This scheduled job is managed by Izzy internally. It is best not to modify it with this command.\n" +
+                                  "Creation syntax:\n" +
+                                  "```\n" +
+                                  $"{_config.Prefix}schedule add {searchString} <date/time>\n" +
+                                  "```";
```
Comment (MercurialPony) [inline Izzy-Moonbot/Modules/ModMiscModule.cs]: Could it be worth replacing the individual strings with a multiline string for better readability?

**Review by MercurialPony — COMMENTED** [review]
Can't say I'm a huge fan of hardcoding logic like the case with the big switch statements here. I'd much rather utilize polymorphism, but that's beyond the scope of this PR for sure

Rest looks good!

**Comment by Ixrec** [comment]
I've always been unsure whether job/action types were better as a polymorphic class hierarchy rather than a single type with an enum field. The switch statements here cannot be easily converted to polymorphism (e.g. they use `_config`), and even if they could `.schedule` still fundamentally needs to know the list of concrete job/action types, and those two points also apply to ScheduleService's logic to execute the jobs, which feel to me like strong arguments in favor of one type with an enum field. Or you could flip that logic and say the real problem is e.g. all of ScheduleService's banner-related code belongs in ScheduledBannerRotationJob instead. So far I haven't come up with a compelling need to do a big refactor in either direction.

**Comment by MercurialPony** [comment]
I haven't had the chance to look at the code closer so it's hard for me to say right now, but either way it's not really a problem so it's nothing critical. I may revisit the idea of refactoring this when I start poking around the code myself.

File: Izzy-Moonbot/Modules/ModMiscModule.cs [hunk 2]
```diff
@@ -525,58 +526,81 @@ public async Task ScheduleCommandAsync([Remainder]string argsString = "")
...
+                        break;
+                    case "ScheduledBoredCommandsJob":
+                        content = $"""
+                            **Bored Commands**
+                            *If no one has posted in `BoredChannel` within the last `BoredCooldown` seconds, this job posts one of the strings in `BoredCommands`, typically a command meant to be executed by Izzy herself or another bot. Then reschedules itself for `BoredCooldown` seconds in the future.*
+                            :warning: This scheduled job is managed by Izzy internally. It is best not to modify it with this command.
+                            Creation syntax:
+                            ```
+                            {_config.Prefix}schedule add {searchString} <date/time>
+                            ```"
```
Comment (MercurialPony) [inline Izzy-Moonbot/Modules/ModMiscModule.cs]: The " might be a typo here

**Review by MercurialPony — APPROVED** [review]

**Pull request merged**